### PR TITLE
Switch OpenShift image mirroring jobs to Prowjobs

### PR DESCRIPTION
**Pull request by petr-muller**
https://github.com/openshift/release/pull/4259 added Prowjob-based versions of image-mirroring jobs running with `--dry-run=true`. The OpenShift job [ran fine](https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/logs/periodic-image-mirroring-openshift/1), so this PR switches the mirroring jobs to Prowjobs entirely. The Prowjobs have the following advantages:

1. Their results and logs can be investigated via [Deck](https://prow.svc.ci.openshift.org/?job=periodic-image*)
2. There is a single Prowjob for each "thing" (openshift, knative, kubefed...) which iterates over its mappings, so setting up mirroring for new version means just adding a mapping file and not a new job.
3. I [documented](https://github.com/openshift/release/tree/master/core-services/quay-mirroring) how this works.

This PR does the following:

- Removes updating the CM from the postsubmit. This is not needed because both mappings and prowjob configs are automatically updated by config-updater.
- Moves the custom mapping regenerating targets from root Makefile into `core-services/quay-mirroring/openshift/`
- Switches the Prowjob to run `@hourly` like the original cronjobs
- Removes all config from the old config location
- Removes watching the old mapping for updates into `image-mirror` CM

/cc @smarterclayton @stevekuznetsov

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@smarterclayton this one will need your express approval

**Comment by petr-muller** [comment]
/hold

Some of the other mirroring jobs have problems with authentization, maybe the secrets are somehow wired wrong. Need to investigate.

**Comment by petr-muller** [comment]
/hold cancel

Resolved, should not affect these jobs (see https://github.com/openshift/release/pull/4306 for details)

File: core-services/quay-mirroring/openshift/Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Regenerate the on cluster image streams from the authoritative mirror
```
Comment (smarterclayton) [inline core-services/quay-mirroring/openshift/Makefile]: Why is this file specific to quay?  We may very well have others, suggest this should be "image-mirroring"

File: cluster/ci/config/mirroring/origin_v3_9 [hunk 2]
```diff
@@ -1,17 +0,0 @@
-docker.io/openshift/origin:v3.9 quay.io/openshift/origin:v3.9 quay.io/openshift/origin-control-plane:v3.9
```
Comment (smarterclayton) [inline cluster/ci/config/mirroring/origin_v3_9]: Why are these deleted?

Comment (smarterclayton) [reply cluster/ci/config/mirroring/origin_v3_9]: All 3.x ones have to be preserved.

File: core-services/quay-mirroring/openshift/Makefile (hunk 1 above)
Comment (petr-muller) [reply core-services/quay-mirroring/openshift/Makefile]: Yeah, why not. I'll rename it.

File: cluster/ci/config/mirroring/origin_v3_9 (hunk 2 above)
Comment (petr-muller) [reply cluster/ci/config/mirroring/origin_v3_9]: They are only removed because I made copies in an earlier PR, the copies are in https://github.com/openshift/release/tree/master/core-services/quay-mirroring/openshift, including the 3.x ones. I'm keeping the copies in sync in this PR.

The only special one is the 3.9 one - that file exists but there was no job that used it. So I made the copy to not follow the mapping name convention, so it is not actually picked by the mirroring job, but we still keep that file.

File: core-services/quay-mirroring/openshift/Makefile (hunk 1 above)
Comment (petr-muller) [reply core-services/quay-mirroring/openshift/Makefile]: Addressed in a followup https://github.com/openshift/release/pull/4401 that needs to merge after this one, because I need to fixup config-updater config too, which needs to happen in a PR separate from changing the content of the tracked files.

File: cluster/ci/config/mirroring/origin_v3_9 (hunk 2 above)
Comment (smarterclayton) [reply cluster/ci/config/mirroring/origin_v3_9]: Ok
